src/Common/src/System/Net/SafeCloseSocket.Unix.cs

```diff
@@ -189,24 +211,37 @@ public static InnerSafeCloseSocket CreateSocket(AddressFamily addressFamily, Soc
                 int pt = (int)protocolType;
 
                 int fd = Interop.libc.socket(af, sock, pt);
+                if (fd != -1)
+                {
+                    // The socket was created successfully; make it non-blocking.
+                    int err = Interop.Sys.Fcntl.SetIsNonBlocking(fd, 1);
+                    if (err != 0)
+                    {
+                        Interop.Sys.Close(fd);
+                        fd = -1;
+                    }
+                }
```

Comment:
`IsNonBlocking` is set by the `SocketPal.SetBlocking`. It does not reflect whether or not the underlying file descriptor is non-blocking. On *nix, we need to emulate Windows' separation of {non,}blocking I/O and overlapped I/O, which we do by setting every underlying socket to non-blocking and emulating blocking I/O as necessary based on the value of `SafeCloseSocket.IsNonBlocking`.